test/models/name_test.rb

```diff
@@ -2881,4 +2888,78 @@ def test_make_sure_names_are_bolded_correctly
+           "Names below genus with quotes should be unregistrable")
+
+    name = Name.new(text_name: "Agaricus pinyonensis",
+                    author: "Isaacs nom. prov.")
+    assert(name.unregistrable?, "Provisional names should be unregistrable")
+
+    name = Name.new(text_name: "Fulvifomes porrectus",
+                    author: "comb. prov.")
+    assert(name.unregistrable?, "Provisional names should be unregistrable")
+
+    name = Name.new(text_name: "Cortinarius calaisopus", author: "ined.")
+    assert(name.unregistrable?, "Unpublished names should be unregistrable")
+
+    name = Name.new(text_name: "Agricales", author: "sensu lato")
+    assert(name.unregistrable?, "Names s.l. should be unregistrable")
+
+    name = Name.new(text_name: "Eukaryota", rank: :Domain)
+    assert(name.unregistrable?, "Domains should be unregistrable")
+
+    name = Name.new(text_name: "Ericales", classification: "Kingdom: _Plantae_")
+    assert(name.unregistrable?,
+           "Taxa outside of Fungi and slime molds should be unregistrable")
+
+    name = names(:coprinus)
+    assert(name.registrable?, "Non-group fungal names should be registrable")
+
+    # Use Protozoa as a rough proxy for slime molds, which are included
+    # fungal nomenclature registries, even though they are not fungi.
+    name = Name.new(text_name: "Myxomycetes", rank: :Class,
+                    classification: "Kingdom: Protozoa")
+    assert(name.registrable?, "Protozoa should be registrable")
+
+    name = Name.new(text_name: "New species", rank: :Species)
+    assert(name.registrable?,
+           "Non-group, non-domain kingdom-less names should be registrable")
+  end
+
+  def test_searchability_in_registry
+    name = Name.new(text_name: "Eukaryota", rank: :Domain)
+    assert(name.unsearchable_in_registry?, "Domains should be unsearchable")
+
+    name = Name.new(text_name: "Ericales", classification: "Kingdom: _Plantae_")
+    assert(name.unsearchable_in_registry?,
+           "Taxa outside of Fungi and slime molds should be unsearchable")
+
+    name = Name.new(text_name: 'Amanita "sp-01"', author: "crypt. temp.")
+    assert(name.unsearchable_in_registry?,
+           "Cryptonyms should be unsearchable")
+
+    name = names(:boletus_edulis_group)
+    assert(name.searchable_in_registry?,
+           "Fungal `groups` can be searchable in registy")
+
+    name = Name.new(text_name: 'Cortinarus "quoted"', rank: :Species)
+    assert(name.searchable_in_registry?,
+           "Names with quote marks can be searchable")
+
+    name = Name.new(text_name: "Agaricus pinyonensis",
+                    author: "Isaacs nom. prov.")
+    assert(name.searchable_in_registry?,
+           "Provisional names can be searchable in registry")
+
+    name = Name.new(text_name: "Myxomycetes", rank: :Class,
+                    classification: "Kingdom: Protozoa")
+    assert(name.searchable_in_registry?,
+           "Protozoa should be searchable in registry")
+  end
```

Comment:
Huge respect for actually providing useful failure messages!  I wish we all would do that.  Would save so much time parsing tests to figure out what it's supposed to be doing and why it's failing!